go/bundle/cmd/sign-bundle/integrityblock.go

```diff
@@ -3,15 +3,92 @@ package main
 import (
 	"crypto"
 	"crypto/ed25519"
+	"encoding/binary"
+	"encoding/hex"
 	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"os"
+
+	"github.com/WICG/webpackage/go/integrityblock"
 )
 
+func openFile(path string) (*os.File, error) {
+	fi, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	return fi, nil
+}
+
+// Returns the length of the web bundle parsed from the last 8 bytes of the web bundle file.
+// https://wicg.github.io/webpackage/draft-yasskin-wpack-bundled-exchanges.html#name-trailing-length
+func readWebBundlePayloadLength(bundleFile *os.File) (int64, error) {
+	// Finds the offset, from which the 8 bytes containing the web bundle length start.
+	_, err := bundleFile.Seek(-8, io.SeekEnd)
+	if err != nil {
+		return 0, err
+	}
+
+	// Reads from the offset to the end of the file (those 8 bytes).
+	webBundleLengthBytes, err := ioutil.ReadAll(bundleFile)
+	if err != nil {
+		return 0, err
+	}
+
+	return int64(binary.BigEndian.Uint64(webBundleLengthBytes)), nil
+}
+
+// Returns either the existing integrity block parsed (not supported in v1) or a newly
+// created empty integrity block. Integrity block preceeds the actual web bundle bytes.
+func obtainIntegrityBlock(bundleFile *os.File) (*integrityblock.IntegrityBlock, int, error) {
```

Comment:
We will need the offset information later in the signing process when we will need to know from which offset point onwards we need to copy the web bundle bytes from. :) We cant only trust to the integrity block length information because we dont know, if it was newly created or reading the existing one.

Reply:
I see. Since current code does not use the second return value, could you add a comment about it? Otherwise someone may "clean up" the code removing it :)

Reply:
I refactored the comment to explain why it's needed. PTAL :)